### PR TITLE
Add instruction to use built-in crl distribution point.

### PR DESCRIPTION
**Pull request by lukasz-lobocki**
#### Name of feature:
Built-in crl distribution point.
#### Pain or issue this feature alleviates:
Alleviates the need to create and publish crl using external tools.
#### Is there documentation on how to use this feature? If so, where?
This is the documentation.
#### In what environments or workflows is this feature supported?
Any
#### Supporting links/other PRs/issues:
https://github.com/smallstep/certificates/issues/1423
https://github.com/smallstep/certificates/issues/206#issuecomment-1530219201

💔Thank you!

### CODE REVIEW

File: step-ca/certificate-authority-server-production.mdx [hunk 1]
```diff
@@ -163,8 +163,7 @@ You can [configure certificate lifetimes](./configuration.mdx#basic-configuratio
 
 <Alert severity="info">
   <div>
-    Short-lived certificates are not a replacement for active revocation using certificate revocation lists (CRL) or Online Certificate Status Protocol (OCSP).
-	Automatic active revocation is not available in <Code>step-ca</Code>, but you can manually manage a CRL for your CA certificates. See below.
+    Short-lived certificates are not a replacement for active revocation using certificate revocation lists (CRL) or Online Certificate Status Protocol (OCSP). They offer an alternative that's easier to deploy and maintain.
```
Comment (hslatman) [inline step-ca/certificate-authority-server-production.mdx]: Strictly not indeed, but maybe a little bit of nuance to it: "are not a full/complete replacement"?

File: step-ca/certificate-authority-server-production.mdx [hunk 2]
```diff
@@ -175,112 +174,173 @@ So do some certificates.
...
+### Consider Active Revocation
 
-The value of a two-tiered PKI is that you can add your root CA certificate to the certificate trust store on all of your nodes, and store your root private key completely offline.
-A leaf certificate signed by the CA always comes in a bundle that contains the intermediate CA certificate alongside the leaf certificate.
-With this bundle, any client that trusts your root CA can verify the complete chain of trust.
+You may require Active Revocation if you need immediate certificate revocation,
+or if you are issuing long-lived certificates.
+For this purpose,
+`step-ca` contains a built-in,
+minimalist Certificate Revocation List (CRL) server.
```
Comment (hslatman) [inline step-ca/certificate-authority-server-production.mdx]: Feels like the better term in this context, but I may be wrong
```suggestion
minimal Certificate Revocation List (CRL) server.
```

File: step-ca/certificate-authority-server-production.mdx [hunk 3]
```diff
@@ -175,112 +174,173 @@ So do some certificates.
...
-      -out ca.crl.pem
-  openssl crl \
-      -inform PEM \
-      -in ca.crl.pem \
-      -outform DER \
-      -out ca.crl
-  ```
+But there are downsides:
+CRL adds a service dependency to your PKI.
+Clients check the CRL endpoint on every new connections,
```
Comment (hslatman) [inline step-ca/certificate-authority-server-production.mdx]:
```suggestion
Clients check the CRL endpoint often,
```

Comment (hslatman) [reply step-ca/certificate-authority-server-production.mdx]: Technically clients could be (and probably are) caching the CRL for a specific duration. The CRL also contains an indication as to when the new CRL can be expected. It's possible it happens earlier, but it does put a limit on how the period that an outdated CRL could be in use. There are also delta CRLs, but I don't think we support that.

OCSP happens on every request, or it gets stapled in the TLS response from a server.

File: step-ca/certificate-authority-server-production.mdx [hunk 4]
```diff
@@ -175,112 +174,173 @@ So do some certificates.
...
+  </div>
+</Alert>
+
+#### When To Use Active Revocation?
+
+The value of a two-tiered PKI is in the decoupling of root and intermediate CAs.
+You can add your root CA certificate to the certificate trust store on all of your nodes,
+and store your root private key completely offline.
+When `step-ca` issues a certificate to a client,
+it comes inside a PEM bundle that contains both the intermediate CA certificate(s) and the end entity certificate.
```
Comment (hslatman) [inline step-ca/certificate-authority-server-production.mdx]: There's a few lines with lowercase intermediate, and a few where it's capitalized

File: step-ca/certificate-authority-server-production.mdx [hunk 5]
```diff
@@ -175,112 +174,173 @@ So do some certificates.
...
+   {
+           "subject": {{ toJson .Subject }},
+           "keyUsage": ["certSign", "crlSign"],
+           "basicConstraints": {
+                   "isCA": true,
+                   "maxPathLen": 0
+           },
+           "crlDistributionPoints":
+                   ["http://ca.example.com/1.0/crl"]
+           }
```
Comment (hslatman) [inline step-ca/certificate-authority-server-production.mdx]: It looks superfluous, but maybe needed some ellipsis or something?
```suggestion
           "crlDistributionPoints": ["http://ca.example.com/1.0/crl"]
```

File: step-ca/certificate-authority-server-production.mdx (hunk 4 above)
Comment (tashian) [reply step-ca/certificate-authority-server-production.mdx]: Yes, sometimes with longer sentences I put one sentence fragment per line. It makes diffs easier to read.
The rendered file puts everything together into a single paragraph. So, the user doesn't see this.

Comment (hslatman) [reply step-ca/certificate-authority-server-production.mdx]: My remark was about `Intermediate CA` vs. `intermediate CA`, and it thus not being consistent

Comment (tashian) [reply step-ca/certificate-authority-server-production.mdx]: Oh sorry, I missed that. Good catch!

**Review by tashian — APPROVED** [review]

**Pull request merged**